r/R/query-engine.R

```diff
@@ -121,11 +119,13 @@ ExecPlan <- R6Class("ExecPlan",
             x
           })
         }
+        target_names <- names(.data$aggregations)
+        for (i in seq_len(length(target_names))) {
+          .data$aggregations[[i]][["name"]] <- .data$aggregations[[i]][["target"]] <- target_names[i]
```

Comment:
If "target" and "name" are always the same, why are we passing them twice? Can't we reuse the same `std::vector<std::string>` in the C++ bindings?

Also, why put target names inside the aggregations elements when we're just going to pull them back out and make a vector in C++? If you pass `target_names` directly, you already have `std::vector<std::string>`.

Reply:
For reference, the equivalent SQL is...

```
SELECT function_1(target_1) as name_1, function_2(target_2) as name_2, ... FROM ... GROUP BY ...
```

Each aggregate has (up to) three parts.  The C++ is changing from passing in three vectors of strings (which is error-prone) to passing in one vectors of structs where each item has up to three parts.

Reply:
Ah got it, you're right, I misread, I see that you're making a vector of structs now, so disregard my last comment.

Reply:
@nealrichardson Since you pointed out the `names` and `target` being the same, in previous implementation they were set to the same value. As far as I understood, the test cases are also following that. Is it wise to affix this or just leave it as it is? I have doubts about which one to select.

Reply:
Hmm...do we have any test case doing something like...
```
mtcars %>% group_by(cyl)
    %>% summarise(
  disp_mean = mean(disp),
  hp_mean = mean(hp)
)
```

I would expect `target` to be `disp` and `name` to be `disp_mean`.  It's always possible we are renaming the columns somewhere else in the generated plan as well in which case `name` is meaningless here.